unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/SceneController.cs

```diff
@@ -390,26 +406,91 @@ private IEnumerator DeferredDecoding()
             {
                 maxTimeForDecode = CommonScriptableObjects.rendererState.Get() ? MAX_TIME_FOR_DECODE : float.MaxValue;
 
-                if (payloadsToDecode.Count > 0)
+                if (chunksToDecode.Count > 0)
                 {
-                    string payload = payloadsToDecode.Dequeue();
-
-                    DecodeAndEnqueue(payload);
+                    if (chunksToDecode.TryDequeue(out string chunk))
+                    {
+                        EnqueueChunk(chunk);
 
-                    if (Time.realtimeSinceStartup - start < maxTimeForDecode)
-                        continue;
+                        if (Time.realtimeSinceStartup - start < maxTimeForDecode)
+                            continue;
+                    }
                 }
 
                 yield return null;
+
                 start = Time.unscaledTime;
             }
         }
+        private void EnqueueChunk(string chunk)
+        {
+            string[] payloads = chunk.Split(new [] { '\n' }, StringSplitOptions.RemoveEmptyEntries);
+            var count = payloads.Length;
+
+            for (int i = 0; i < count; i++)
+            {
+                bool availableMessage = sceneMessagesPool.TryDequeue(out QueuedSceneMessage_Scene freeMessage);
+
+                if (availableMessage)
+                {
+                    EnqueueSceneMessage(Decode(payloads[i], freeMessage));
+                }
+                else
+                {
+                    EnqueueSceneMessage(Decode(payloads[i], new QueuedSceneMessage_Scene()));
+                }
+            }
+        }
+        private async UniTask WatchForNewChunksToDecode(CancellationToken cancellationToken)
+        {
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                try
+                {
+                    if (chunksToDecode.Count > 0)
+                    {
+                        await TaskUtils.Run( () => ThreadedDecodeAndEnqueue(cancellationToken), cancellationToken: cancellationToken);
```

Comment:
This is going to get a new thread from the pool for each decoding request. The thread pool wouldn't get dried out if we do this? I think we should aim at using the same worker thread for correctness.